src/components/Toast/Toast.test.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import Toast from './index';
+
+describe('Toast component', () => {
+  test('renders with children', () => {
+    render(<Toast type="warning">Hello World</Toast>);
+
+    expect(screen.getByText('Hello World')).toBeTruthy();
+  });
+
+  test('renders with default classname', () => {
+    render(<Toast type="success">Test</Toast>);
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('Toast');
+  });
+
+  test('renders with success classname', () => {
+    render(<Toast type="success">Test</Toast>);
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('Toast--success');
+  });
+
+  test('renders with warning classname', () => {
+    render(<Toast type="warning">Test</Toast>);
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('Toast--warning');
+  });
+
+  test('renders with passed classname', () => {
+    render(
+      <Toast type="success" className="test">
+        Test
+      </Toast>,
+    );
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('test');
+  });
+
+  test('renders with passed--success classname', () => {
+    render(
+      <Toast type="success" className="test">
+        Test
+      </Toast>,
+    );
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('test--success');
+  });
+
+  test('renders with passed--warning classname', () => {
+    render(
+      <Toast type="warning" className="test">
+        Test
+      </Toast>,
+    );
+    const el = screen.getByTestId('Toast');
+
+    expect(el.className).toContain('test--warning');
+  });
+});
```

Comment:
add tests here that tests for the `IconType.thumbsUp`, and `IconType.alertCircleOutline`.

One way you can do that is within your test file, import the two icons:

`
import AlertCircleOutlineIcon from 'mdi-react/AlertCircleOutlineIcon';
import ThumbsUpIcon from 'mdi-react/ThumbsUpIcon';
`

and then for the `success` icon test, something like `expect(ThumbsUpIcon).toBeEtcEtc`. Something along those lines

Reply:
I've been trying to figure this out for a while, but I can't find the method to use. I found this [stackoverflow question](https://stackoverflow.com/questions/60041468/in-react-testing-library-how-do-i-check-that-a-child-component-is-rendered) which is related but I don't think it is applicable here. Mostly I find that this way of testing is against the RTL principles

Reply:
hmm @AbhayAysola the question in the stack overflow is what I had in mind, but I see that that is for Enzyme (and like the answer says, RTL doesn't encourage that)

maybe you can just pass `data-testid` to the various icons, and just check for those.

Thank you @AbhayAysola for doing the research though!

Reply:
No problem, happy to help! I'll add classNames to the different icons and check for those. @angle943